### PR TITLE
Add initial LIT tests configuration

### PR DESCRIPTION
**Pull request by kuhar**
Allow to add future regression tests.
Add a test that ensures that all tools are available and can be executed.

### CODE REVIEW

**Comment by kuhar** [comment]
@JaxLinAMD @dstutt @JonCampbell407 @grgrebe-amd @trenouf  Please take a look

**Comment by amdvlk-admin** [comment]
### Test summary for commit 664b1a4aab777c28c3b66e18bf2939f665cbe4ee
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev c2c373c45859f58932077a10be8d6db9a9d8432f</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1d0178f3d37675e22bf1a16267d1d070fcb8d720</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/224697)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26847/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25436/56232 (45.2%)</li>
<li>Failed: 0/56232 (0.0%)</li>
<li>Skipped: 30796/56232 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26862/56155 (47.8%)</li>
<li>Failed: 0/56155 (0.0%)</li>
<li>Skipped: 29293/56155 (52.2%)</li>
</ul></details>


</ul></details>

File: tools/cache_creator/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.py.in
+    ${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg.py
+    MAIN_CONFIG
+    ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
+)
+
+# You can execute test by building the `check-cache-creator` target, e.g., `ninja check-cache-creator`.
+add_lit_testsuite(check-cache-creator "Running the XGL cache creator regression tests"
+    ${CMAKE_CURRENT_BINARY_DIR}
+    DEPENDS cache-creator amdllpc FileCheck
```
Comment (dstutt) [inline tools/cache_creator/test/CMakeLists.txt:49]: I think you might need "spvgen", "count" and "not" in this depends line too.

**Review by dstutt — COMMENTED** [review]
On the whole this LGTM - but there's a minor issue if you try and do this from a fresh build. See comment.

**Comment by kuhar** [comment]
V1: Add missing dependencies to the `check-cache-creator` target.

**Comment by kuhar** [comment]
Thanks @dstutt, I hadn't run a fresh build and didn't realize `spvgen` was missing. `count` and `not` are not used at the moment, but I expect them to be required at some point added them to the list as well.

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by Flakebi** [comment]
Do we need spvgen to test the cache creator?
I’m not sure what kind of unit tests you have in mind, as long as they use SPIR-V and not glsl, spvgen is not needed.

**Comment by kuhar** [comment]
V2: Simplify the placeholder test case. Pass a `gfxip` valid version to amdllpc.

**Comment by kuhar** [comment]
@Flakebi 
> Do we need spvgen to test the cache creator?
> I’m not sure what kind of unit tests you have in mind, as long as they use SPIR-V and not glsl, spvgen is not needed.

I think we also need it for spvams files. When I don't specify it, it fails with `ERROR: Failed to load SPVGEN -- cannot assemble SPIR-V assembler source`.
I don't have the full intended test suit at hand, but I think that being able to compose tests out of glsl files might be useful in the future -- it should be much more concise to specify all shader inputs in a single file in this format (with [split-file](https://llvm.org/docs/TestingGuide.html#extra-files). However, if we can assemble spvasm files without spvgen, we don't have to add spvgen until we actually need glsl support.

**Comment by Flakebi** [comment]
Interesting, I would have expected it to work. If you want to use glsl in the future anyway, then it makes sense to add it now with all the other changes, so sounds good to me.

**Review by dstutt — APPROVED** [review]
LGTM

**Review by JonCampbell407 — APPROVED** [review]

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3e630227fd0dd01d7306e8dd92cdb114ac311c6d
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>LLPC: dev c4da4ae802896acdb8334db7b4a88a328e37d027</li>
<li>PAL: dev 1e61f9ab6eb2041c73d7e7bd8f2021fa05a4bdd8</li>
<li>SPVGEN: dev 7c1cf34dd7eb4d3a6e6652c26292eef1bfcbcaf0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev bcd2fe473ee79c4907f2136ab963c1a3676c9db5</li>
</ul></details>

<details><summary>CTS tests (Failed: 336/224697)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26763/56155 (47.7%)</li>
<li><details><summary>Failed: 84/56155 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26763/56155 (47.7%)</li>
<li><details><summary>Failed: 84/56155 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29308/56155 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25352/56232 (45.1%)</li>
<li><details><summary>Failed: 84/56232 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 30796/56232 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26778/56155 (47.7%)</li>
<li><details><summary>Failed: 84/56155 (0.1%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.device_init.create_instance_device_intentional_alloc_fail
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_storage_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_large
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_uniform_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_storage_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.buffer_view_uniform_r8g8b8a8_unorm
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.command_pool_transient
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.compute_pipeline
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_pool_free_descriptor_set
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_empty
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.descriptor_set_layout_single
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_group
Stack trace: Crash

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.device_memory_small
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.event
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence
Stack trace: Fail

FAILURE: dEQP-VK.api.object_management.alloc_callback_fail.fence_signaled
Stack trace: Fail
...
```
</p></details></li>

<li>Skipped: 29293/56155 (52.2%)</li>
</ul></details>


</ul></details>

**Pull request merged**